app/src/main/java/com/puzzlebench/clean_marvel_kotlin/presentation/mvp/CharacterPresenter.kt

```diff
@@ -28,19 +29,48 @@ class CharacterPresenter constructor(view: CharacterView,
     }
 
     private fun requestGetCharacters() {
-        val subscription = getCharacterServiceUseCase.invoke().subscribeOn(Schedulers.io()).observeOn(AndroidSchedulers.mainThread()).subscribe({ characters ->
+        val subscription = getCharacterServiceUseCase.invoke()
+                .subscribeOn(Schedulers.io())
+                .observeOn(AndroidSchedulers.mainThread())
+                .subscribe({ characters ->
             if (characters.isEmpty()) {
                 view.showToastNoItemToShow()
             } else {
                 saveCharacterRepositoryUseCase.invoke(characters)
                 view.showCharacters(characters)
             }
             view.hideLoading()
-
-        }, { e ->
+            view.showFAB()
+                }, { e ->
             view.hideLoading()
+            view.showFAB()
             view.showToastNetworkError(e.message.toString())
         })
         subscriptions.add(subscription)
     }
+
+    fun onClickRefreshFAB() {
+        characters = emptyList()
+        view.hideFAB()
+        view.showCharacters(characters)
+        view.showLoading()
+        requestGetCharacters()
+    }
+
+    fun onClickDatabaseFAB() {
+        view.hideFAB()
+        characters = emptyList()
+        view.showCharacters(characters)
+        view.showLoading()
+        characters = getCharacterRepositoryUseCase.invoke()
+        view.showCharacters(characters)
```

Comment:
you can use directly 
view.showCharacters(getCharacterRepositoryUseCase.invoke())

Reply:
done